oauth2/main.go

```diff
@@ -357,6 +366,40 @@ func generateString() (string, error) {
 	return string(b), nil
 }
 
+func promptForSelect(ctx context.Context, g *gptscript.GPTScript) (authType, error) {
+	fieldName := "Authentication Method"
+
+	fields := gptscript.Fields{gptscript.Field{Name: fieldName, Description: "The authentication method to use for this tool.", Options: []string{string(authTypePAT), string(authTypeOAuth)}}}
+	sysPromptIn, err := json.Marshal(sysPromptInput{
+		Message: "This tool has personal access token (PAT) and OAuth support. Select the authentication method you would like to use for this tool.",
+		Fields:  fields,
+	})
+	if err != nil {
+		return "", fmt.Errorf("promptForSelect: error marshalling sys prompt input: %w", err)
+	}
+
+	run, err := g.Run(ctx, "sys.prompt", gptscript.Options{
+		Input: string(sysPromptIn),
+	})
+	if err != nil {
+		return "", fmt.Errorf("promptForSelect: failed to run sys.prompt: %w", err)
+	}
+
+	out, err := run.Text()
+	if err != nil {
+		return "", fmt.Errorf("promptForSelect: failed to get prompt response: %w", err)
+	}
+
+	m := make(map[string]string)
+	if err = json.Unmarshal([]byte(out), &m); err != nil {
+		return "", fmt.Errorf("promptForSelect: failed to unmarshal prompt response: %w", err)
+	}
+
+	authType := authType(m[fieldName])
```

Comment:
```suggestion
	selectedAuthType := authType(m[fieldName])
```
Optional. I was just confused by the variable name lol, as it is also a custom type